Sources/ProjectSpec/Dependency.swift

```diff
@@ -32,11 +32,18 @@ public struct Dependency: Equatable {
         self.implicit = implicit
         self.weakLink = weakLink
     }
+    
+    public enum CarthageLinkType: String {
+        case dynamic
+        case `static`
+        
+        public static let `default` = dynamic
+    }
 
     public enum DependencyType: Equatable {
         case target
         case framework
-        case carthage(findFrameworks: Bool?)
+        case carthage(findFrameworks: Bool?, linkType: CarthageLinkType = .default)
```

Comment:
I'm not sure if we are dropping lower Swift compiler versions here yet, so we shouldn't use a default associated value.

Reply:
reverted. https://github.com/yonaskolb/XcodeGen/pull/688/commits/14a249d7c306db3618333c34f6a7441916c24242